File: app/src/main/java/ru/nehodov/weatherforecast/utils/CurrentToDailyConverter.java
package ru.nehodov.weatherforecast.utils;

import ru.nehodov.weatherforecast.entities.Current;
import ru.nehodov.weatherforecast.entities.Daily;
import ru.nehodov.weatherforecast.entities.FeelsLike;
import ru.nehodov.weatherforecast.entities.Temp;

public class CurrentToDailyConverter {

    public static Daily convert(Current current) {
        Temp temp = new Temp(current.getTemp(), current.getTemp());
        FeelsLike feelsLike = new FeelsLike(current.getFeelsLike(), current.getFeelsLike());
        return new Daily(
                current.getId(),
                current.getSunrise(),
                temp,
                feelsLike,
                current.getUvi(),
                current.getPressure(),
                current.getClouds(),
                current.getDateTime(),
                current.getSunset(),
                current.getWeather(),
                current.getHumidity(),
                current.getWindSpeed()
        );
    }

}

File: app/src/main/java/ru/nehodov/weatherforecast/utils/CurrentToDailyConverter.kt
package ru.nehodov.weatherforecast.utils

import ru.nehodov.weatherforecast.entities.Current
import ru.nehodov.weatherforecast.entities.Daily
import ru.nehodov.weatherforecast.entities.FeelsLike
import ru.nehodov.weatherforecast.entities.Temp

object CurrentToDailyConverter {

    fun convert(current: Current): Daily {
        val temp = Temp(current.temp, current.temp)
        val feelsLike = FeelsLike(current.feelsLike, current.feelsLike)
        return Daily(
            current.id,
            current.sunrise,
            temp,
            feelsLike,
            current.uvi,
            current.pressure,
            current.clouds,
            current.dateTime,
            current.sunset,
            current.weather,
            current.humidity,
            current.windSpeed
        )
    }
}